python/src/iceberg/catalog/base.py

```diff
@@ -0,0 +1,88 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Tuple
+
+import mmh3  # type: ignore
+
+
+class Namespace:
+    """A namespace in a Catalog
+
+    Example:
+        >>> from iceberg.catalog.base import Namespace
+        >>> namespace = Namespace("foo", "bar", "baz")
```

Comment:
just curious about, for namespace, are we anticipating something like `iceberg.datdabase.table` and potentially the 4th level with metadata tables such as `partitions` or `files`? or is it being used different?

Trying to see if we shall build some assumptions into length of the namespace, too long or too short wont make any sence?

Reply:
This is a good question. My opinion is that it's ok to nest the source code files as much as logically makes sense and makes navigating the code base easier. We always have an option of using __all__ to bubble items up to the sub-module level which I think we should.